persistence/src/main/scala/com/evolutiongaming/akkaeffect/persistence/Recovering.scala

```diff
@@ -30,10 +30,27 @@ trait Recovering[F[_], S, E, +A] {
     journaller: Journaller[F, E],
     snapshotter: Snapshotter[F, S],
   ): Resource[F, A]
+
+  /** Called when recovery completed, resource will be released upon actor termination
+    *
+    * @see
+    *   [[akka.persistence.RecoveryCompleted]]
+    */
+  def completed(context: Recovering.RecoveryContext[F, S, E]): Resource[F, A] =
+    completed(context.seqNr, context.journaller, context.snapshotter)
 }
 
 object Recovering {
 
+  /** Context containing information about recovery and provides access to journaller and snapshotter
+    */
+  trait RecoveryContext[F[_], S, E] {
+    def seqNr: SeqNr
+    def journaller: Journaller[F, E]
+    def snapshotter: Snapshotter[F, S]
+    def recoveredFromPersistence: Boolean
```

Comment:
I wonder if ADT would be more expressive here?

```scala
sealed trait RecoveredFrom
case object Persistence
case object StateTransfer
// ...
def recoveredFrom: RecoveredFrom
```

Reply:
> case object StateTransfer

that's what I tried to hide :D

Reply:
I've never liked the boolean flags - it is always hard to understand, what the "other" case means, and sometimes developers add second and third boolean...